challenges/Sudoku-BackTracking.md

```diff
@@ -0,0 +1,129 @@
+# Solving a Sudoku puzzle using backtracking
+
+## What is Backtracking?
+Backtracking is an algorithmic-technique for solving problems recursively by trying to build a solution incrementally, one piece at a time, removing those solutions that fail to satisfy the constraints of the problem at any point of time (by time, here, is referred to the time elapsed till reaching any level of the search tree).
+More [here](https://www.geeksforgeeks.org/backtracking-introduction/).
+
+## How does this help in solving sudoku?
+The algorithm tries the first valid number it can put in the first empty square it finds on the grid. After this it recursively calls the function and it tries the first valid number on the next empty square.
+If we encounter an empty square where no number is valid then the algorithm goes back to the previous recursive state, finds the next valid number in the previous step and builds upon that number.
+This might not seem intuitive at the first go but you could find an amazing visualisation video on the topic [here](https://www.youtube.com/watch?v=_vWRZiDUGHU).
+
+## C++ Code
```

Comment:
```suggestion
## Code
```

Reply:
I think keeping C++ here would make more sense since the code uses C/C++ syntax. I could make it C/C++ code though.

Reply:
Oh right cause you've added this in challenges folder. Then we should probably add c++/cpp in the name of the file instead of here.

Reply:
The file's name is already too big. Are you sure you want to add C/C++ in the file itself?

Reply:
I think it's not that big. You can add c++ or cpp.